src/core/classes/TransactionTask.ts

```diff
@@ -0,0 +1,124 @@
+import { ContractWrapper } from "./contract-wrapper";
+import { BigNumber, CallOverrides, ContractTransaction } from "ethers";
+import { ContractEncoder } from "./contract-encoder";
+import { GasCostEstimator } from "./gas-cost-estimator";
+import { TransactionResult } from "../types";
+
+class TransactionTaskBuilder {
+  public contractWrapper: ContractWrapper<any>;
+  public functionName: string;
+  public args: any[] = [];
+  public overrides: CallOverrides | undefined;
+
+  constructor(contractWrapper: ContractWrapper<any>, functionName: string) {
```

Comment:
Would it be beneficial to use/at least support the argument object pattern which tends to be common in JS vs the builder pattern. We had builder before with claim conditions and moved away from it - these two aren't as complex (this comment applies to both `TransactionTaskBuilder` and `TransactionTask`) but could be more convenient

Reply:
I see what you mean - I kinda gravitated towards a builder pattern mainly coz im used to it. Might not be the JS way. I'll play with an object as argument, prob less code too